### PR TITLE
Present the list of dependencies for each repository

### PR DESCRIPTION
**Pull request by danlivings-dxw**
Being able to see dependency information at a glance will be one of the primary benefits of Towtruck. This PR presents a first iteration for displaying this information, which is sourced from the Renovate Dependency Dashboard (if Renovate is configured to create one for a given repository). It does not currently display information about new releases or expected end-of-life dates.

This PR implements issue #17.

Below is a preview of these changes:

1. When the dependency list is collapsed: image
2. When the dependency list is expanded: image

### CODE REVIEW

File: renovate/dependencyDashboard.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    this.name = name;
+    this.version = version;
+  }
+}
+
+export const getDependenciesForRepo = ({ octokit }, repo) => {
+  return octokit
+    .request(repo.issuesUrl)
+    .then((response) => {
+      const dependencies = response
```
Comment (richpjames) [inline renovate/dependencyDashboard.js:12]: I think this would be easier to test if we only tested the functionality in this `then` block rather than the `octokit` code as well. We don't own the `octokit` code.

File: renovate/dependencyDashboard.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        .data
+        .find((issue) => issue.user.login === "renovate[bot]" && issue.pull_request === undefined)
+        ?.body
+        ?.split(/\r?\n|\r|\n/g)
+        ?.map((line) => {
```
Comment (richpjames) [inline renovate/dependencyDashboard.js:17]: I find this function quite hard to parse. I think if I had to jump into this repo to do maintenance it would take me quite a while to work out what is going on. 
Do you think it would be clearer if it was broken down into smaller functions that can be named to demonstrate to the reader what is being done?

File: renovate/dependencyDashboard.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+          const match = line.match(/`(\S+?) (.*)`/);
```
Comment (richpjames) [inline renovate/dependencyDashboard.js:18]: I think it would be worth extracting this regex to a const so it can be named to help aid the reader's understanding.

**Review by richpjames — COMMENTED** [review]
I'm a bit concerned that the Renovate Dependency Dashboard will give us the information that's needed to keep these repos up to date.  A few reasons: 
- Most of dxw's repos don't have the dashboard, I had a look at 200 and only 3 of them had a dashboard.
- Of the dashboards that do it exist, it doesn't seem able to show the version for a lot of the dependencies. For example for a lot Ruby Gems it just shows `undefined`: 
image  
- We are using regex to parse the body of the Renovate dashboard. If the format of this body changed then the dashboard would break.  
This dashboard data could be useful to guide decision but I'm not sure it would be the primary source of information.

**Comment by danlivings-dxw** [comment]
@richpjames I think Renovate is probably our best bet.

> * Most of dxw's repos don't have the dashboard, I had a look at 200 and only 3 of them had a dashboard.

There could be an argument to be made here that setting up Renovate to automatically update minor dependency versions and automatically create PRs for major versions would be an important step to reduce the manual workload in maintaining our repos.

> * Of the dashboards that do it exist, it doesn't seem able to show the version for a lot of the dependencies. For example for a lot Ruby Gems it just shows `undefined`:

Looking at some `Gemfile`s where Renovate reports the version as `undefined`, there's no version specified (for example [here](https://github.com/dxw/ejudiciary-homepage-replica/blob/master/Gemfile)). I don't know enough about Ruby to know if there is even a way to know what versions these will resolve to without actually building and running the project.

> * We are using regex to parse the body of the Renovate dashboard. If the format of this body changed then the dashboard would break. This dashboard data could be useful to guide decision but I'm not sure it would be the primary source of information.

Parsing them from the issue body text isn't ideal. The alternative would be to use the Renovate CLI tool, but then we'd need Towtruck to clone every repository and run it. Directly parsing dependency information from build scripts to me is a non-starter as at that point we're building a worse Renovate that will require manual intervention if we want Towtruck to be useful for a project in a new language.

File: renovate/dependencyDashboard.js (hunk 1 above)
Comment (danlivings-dxw) [reply renovate/dependencyDashboard.js]: I've updated the tests.

File: renovate/dependencyDashboard.js (hunk 2 above)
Comment (danlivings-dxw) [reply renovate/dependencyDashboard.js]: Done.

File: renovate/dependencyDashboard.js (hunk 3 above)
Comment (danlivings-dxw) [reply renovate/dependencyDashboard.js]: I've put both of the regexes into `const`s with a comment for each explaining their purpose.
